nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/file/classloader/ClassLoaderUtils.java

```diff
@@ -52,23 +86,29 @@ public static ClassLoader getCustomClassLoader(String modulePath, ClassLoader pa
                     isUrl = false;
                 }
                 if (!isUrl) {
-                    File modulePath = new File(modulePathString);
+                    try {
+                        File modulePath = new File(modulePathString);
 
-                    if (modulePath.exists()) {
+                        if (modulePath.exists()) {
 
-                        additionalClasspath.add(modulePath.toURI().toURL());
+                            additionalClasspath.add(modulePath.toURI().toURL());
 
-                        if (modulePath.isDirectory()) {
-                            File[] files = modulePath.listFiles(filenameFilter);
+                            if (modulePath.isDirectory()) {
+                                File[] files = modulePath.listFiles(filenameFilter);
 
-                            if (files != null) {
-                                for (File jarFile : files) {
-                                    additionalClasspath.add(jarFile.toURI().toURL());
+                                if (files != null) {
+                                    for (File jarFile : files) {
+                                        additionalClasspath.add(jarFile.toURI().toURL());
```

Comment:
I see what you mean, I'll add that check to see if it is a directory or file and handle accordingly